src/wp-includes/block-editor.php

```diff
@@ -243,6 +243,41 @@ function get_block_editor_settings( $editor_name, $custom_settings = array() ) {
 		$custom_settings
 	);
 
+	$editor_settings['__experimentalFeatures'] = WP_Theme_JSON_Resolver::get_merged_data( $editor_settings )->get_settings();
+
+	// These settings may need to be updated based on data coming from theme.json sources.
+	if ( isset( $editor_settings['__experimentalFeatures']['color']['palette'] ) ) {
+		$editor_settings['colors'] = $editor_settings['__experimentalFeatures']['color']['palette'];
+	}
+	if ( isset( $editor_settings['__experimentalFeatures']['color']['gradients'] ) ) {
+		$editor_settings['gradients'] = $editor_settings['__experimentalFeatures']['color']['gradients'];
```

Comment:
I wonder if we could remove the old settings e.g: $editor_settings['gradients'] and all this code ? The downside is that plugins filtering the old settings may not work as expected but even if we keep the old settings plugins filtering the old settings will still not work right? Because now the editor will rely on __experimentalFeatures and not on the old settings?


What are the cases that make us need to keep the old settings?
Should we write some dev notes related to the new settings format?

Reply:
See the [comment above](https://github.com/WordPress/wordpress-develop/pull/1262#discussion_r634588154)) about retrofitting these values for existing plugins. I pointed to a [previous conversation we had about this](https://github.com/WordPress/gutenberg/issues/30082#issuecomment-816679803). We landed on retrofitting those values for existing code to work. We could add a deprecation message or something like that if we wanted to remove them in subsequent releases, but, given that `__experimentalFeatures` is still experimental I wouldn't remove them now.

Reply:
So for the top level settings our code will use the existing settings and for the block level settings __experimentalFeatures is not used?

Reply:
The `useSetting` hook uses the `settings.__experimentalFeatures` (with fallbacks for some cases) for everything. However, there may be code from plugins, etc that don't use this hook yet but instead directly inspect the store for `settings.colors`, etc. It's for those cases that we need to retrofit the new to the existing ones.

Reply:
So a plugin is changing the color palette by filtering $editor_settings['colors'] will stop working because now the editor will use settings.__experimentalFeatures in most cases?

Reply:
Closing as per https://github.com/WordPress/wordpress-develop/pull/1262#discussion_r634596983